torch/distributed/algorithms/_comm_hooks/allreduce_hook.py

```diff
@@ -0,0 +1,59 @@
+import torch
+import torch.distributed as dist
+from torch.distributed import distributed_c10d
+
+
+class AllReduceState(object):
+    r"""
+    Stores parameters, needed to perform ``all_reduce`` algorithm
+    within a communication hook.
+
+    Args:
+        process_group (ProcessGroup): The process group to be used for all-reduce.
+        world_size (int): The number of nodes in a process group.
```

Comment:
There may be multiple workers per node (e.g. 8 GPUs per 1 node in a multi-node cluster).
```suggestion
        world_size (int): The number of workers in a process group.
```